includes/modules/order_total/ot_coupon.php

```diff
@@ -313,23 +313,21 @@ function collect_posts() {
         // JTD - end of handling coupon product restrictions
 
         $date_query=$db->Execute("select coupon_start_date from " . TABLE_COUPONS . "
-                                  where coupon_start_date <= now() and
-                                  coupon_code='" . zen_db_prepare_input($dc_check) . "'");
+                                  where coupon_code='" . zen_db_prepare_input($dc_check) . "' LIMIT 1");
 
-        if ($date_query->RecordCount() < 1 ) {
-          $messageStack->add_session('redemptions', TEXT_INVALID_STARTDATE_COUPON . ' ' . ($dc_link_count ==0 ? $dc_link : ''),'caution');
+        if (date("Y-m-d H:i:s") < $date_query->fields['coupon_start_date']) {

```

Comment:
I'm not sure I understand why you're doing another date-comparison here when it was just done 2 lines earlier in the database query so the query results are only those within a suitable date range.

Reply:
not sure what you mean. I just substituted the sql checks that were there already for the start and expire dates. You mean you want to merge to a single query with a single error message for an invalid date?

Reply:
No, I mean, I don't see why you made a change at all. There's no need to add a PHP date comparison when it's already done in the sql.

Reply:
ah, it's to provide date information in the message. Previously it would just return no result, now it returns the date for use in the message.